aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -127,6 +127,31 @@ struct WelfordOps {
   }
 };
 
+template <typename acc_t, typename data_t>
```

Comment:
This (having `data_t`) should only be needed if we allow FP16 support for CPU kernel. Currently, `prod` does not support `Half` precision (`sum` and `nansum` use FP32 if given FP16, not sure why this promotion is needed though).

I'll remove `data_t` for now. (small todo, don't want to break tests flow right now).

Reply:
Sum probably accumulates in fp32 instead of fp16 to avoid overflow.